src/ray/raylet/node_manager.cc

```diff
@@ -460,6 +482,10 @@ void NodeManager::ProcessClientMessage(
 
       // Since some resources may have been released, we can try to dispatch more tasks.
       DispatchTasks();
+    } else {
```

Comment:
There were a bunch of changes to this part of the code in #2362. I think now the right way to get the driver ID is shown in 

https://github.com/ray-project/ray/blob/ccb9a27393022b3e44fa7e073128c752ef119664/src/ray/raylet/node_manager.cc#L522-L526